### DISCUSSION TITLE
Read-only state for Combobox/Select

### DISCUSSION THREAD

**Discussion by PoojMenon** [twilio-labs/paste#1682] [Q&A] [answered]
I have a form consisting of input fields and a Select. Based on the user's permissions, these fields become read-only when the user doesn't have permission to edit. 

However, I just noticed that Select/Combobox doesn't have a read-only state which makes the form look odd with a mix of readonly/disabled fields. The disabled state also makes the information in the field difficult to read and the intent of the page is to still allow the user to see that information even if they can't edit.

![image](https://user-images.githubusercontent.com/72031986/126291707-199d49e3-0cdf-4a6c-9dbb-5f3516c5fcf5.png)

**Comment by TheSisb** [comment, marked as answer]
Hi @PoojMenon,

Would you be able to use a readonly input instead of a combobox in that case or do you want the user to still be able to open the dropdown? I'm not sure allowing the dropdown to open would be helpful in terms of UX. If it's a matter of having an easy way to display a readonly input with the dropdown arrow, I also wonder if the arrow would cause confusion.  I'm not sure though so I'll check in with designers on my team.

**Reply by TheSisb** [reply]
Checked in with my team.  An alternative approach you may also consider is showing the text as plain text, like in the [Object Details pattern](https://paste.twilio.design/patterns/object-details/#tabbed-details).  Especially if permissions are set at the user level. I'm not sure if a user without permissions needs to know that those are form fields.

**Reply by PoojMenon** [reply]
That's super helpful. I think the Object details pattern makes more sense in this case.
